pyccl/halos/profiles_2pt.py

```diff
@@ -111,6 +124,9 @@ def fourier_2pt(self, cosmo, k, M, a, prof, *, prof2=None):
                 is desired. If ``None``, the assumption is that you want
                 an auto-correlation. Note that only auto-correlations
                 are allowed in this case.
+            diag (bool): If True, both halo profiles depend on the same k. If
+            False, they will depend on k and k', respectively and we will
+            approximate <uk uk'> to <uk><uk'>. Default True.
```

Comment:
Meh, we should actually modify `fourier_variance` in the HOD profile to do this properly, rather than approximating.

Reply:
Also, isn't there a similar thing for the CIB?

Reply:
Should I modify this in this PR? Regarding CIB, I didn't touch it. Should I add the diag argument too?

Reply:
Humm, let's leave it as it is. Can you add a TODO comment somewhere?

Reply:
But you should do a similar thing for the CIB profile here: https://github.com/LSSTDESC/CCL/blob/a427bddaabdacb3d43188d7c3b4b4f1fcd1f0b78/pyccl/halos/profiles_2pt.py#L134

Reply:
Done